administrator-guides/admin/file-upload/README.md

```diff
@@ -0,0 +1,237 @@
+Use this setting to enable or restrict file uploads on Direct Messages
+
+### Maximum File Upload Size (in bytes)
+
+Use this setting to define the maximum allowed file size for uploads.
+
+> Set it to -1 to remove the file size limitation.
+
+### Accepted Media Types
+
+Use this setting to define what types of files the users will be able to upload.
+
+> Comma-separated list of media types. Leave it blank for accepting all media types.
+
+### Protect Uploaded Files
+
+- **Yes**: Only authenticated users will have access to the uploaded files.
+- **No**: If in possession of the uploaded file's URL, unauthenticated users will be able to access and download said files.
+
+### Storage Type
+
+- **GridFS**: Uses the GridFS storage type
+- **AmazonS3**: Uses the [Amazon S3](../file-upload/#amazon-s3) storage type
+- **Google Cloud Storage**: Uses the [Google Cloud Storage](../file-upload/#google-cloud-storage) storage type
+- **WebDAV**: Uses the [WebDav](../file-upload/#webdav) storage type
+- **FileSystem**: Uses the [FileSysyem](../file-upload/#filesystem) storage type
+
+- - -
+
+## Amazon S3
+
+You can find a more detailed guide on how to set your Amazon S3 server [here](/administrator-guides/file-upload/amazon-s3/).
+
+### Access Key
+
+Your Amazon Access Key ID (optional)
+
+### Secret Key
+
+Your Amazon Secret Access Key (optional)
+
+> **Note**: If no `AWSAccessKeyId` and `AWSSecretAccessKey` are set, the underlying AWS SDK will fallback to trying to retrieve credentials from the [usual locations](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-credentials-node.html).
+
+### Acl
+
+(Optional)
+
+A _Canned ACL_ configuration (see a full list [here](http://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl))
+
+### Bucket name
+
+The bucket name you've created on Amazon S3
+
+### Bucket URL
+
+(Optional)
+
+Override URL to which files are uploaded. This URL will also be used for downloads unless a CDN is given
+
+### CDN Domain for Downloads
+
+(Optional)
+
+If you had set a CDN for your bucket, put it here
+
+<!-- ### Force Path Style
+
+- **Yes**: _FILL HERE_
+- **No**: _FILL HERE_ -->
+
+### Proxy Avatars
+
+Proxy avatar file transmissions through your server instead of direct access to the asset's URL
+
+### Proxy Uploads
+
+Proxy upload file transmissions through your server instead of direct access to the asset's URL
+
+### Region
+
+(Optional)
+
+If you have created your bucket on region different from the US Standard, you **have** to set their code here (see a list of region codes [here](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions))
+
+### Signature Version
+
+Identifies the version of AWS Signature that you want to support for authenticated requests.
+
+### URLs Expiration Timespan
+
+Time after which Amazon S3 generated URLs will no longer be valid (in seconds). If set to less than 5 seconds, this field will be ignored.
```

Comment:
```suggestion
Time after which Amazon S3 generated URLs will no longer be valid (in seconds). If set to less than 5 seconds, this field will be ignored.

> If you run into issues with files not loading consistently for some users located further from server location.  Increasing this might help
```